### PR TITLE
Add SimplenoteIntents extension to app

### PR DESCRIPTION
**Pull request by charliescheer**
### Fix
This PR lays the ground work for adding intent configuration widgets into Simplenote.

As compared to the new note widget, which displays no data, to be able to setup widgets that can be configured by the user we need to create another extension for the app that defines an Intent configuration.  This PR sets that up.

### Test
There isn't much to test with this PR, but you can confirm that the option to configure a widget is there...
There is a placeholder for the note preview widget that was added to be able to confirm the intents were working, so to test this pr, make sure that you have configuration options.  To do that.
1. run the app on device or in the sim to install the widgets
2. On the home screen long press till the home screen editor appears, tap on the plus
3. find simplenote and select the note preview widget
4. Once it the widget is added to your home screen, while home screen edit is enabled, tap on note preview widget.
5. you will see an option to select the note.  Tap on that and you will be given four choices Note 1-4
6. select one and see that the text appears in the widget.

### CODE REVIEW

File: Simplenote.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4648,6 +4753,181 @@
...
+				INFOPLIST_FILE = SimplenoteIntents/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 14.4;
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+					"@executable_path/../../Frameworks",
+				);
+				MTL_ENABLE_DEBUG_INFO = NO;
+				MTL_FAST_MATH = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = com.codality.NotationalFlow.SimplenoteIntents;
```
Comment (jleandroperez) [inline Simplenote.xcodeproj/project.pbxproj]: We've got:
- **App:** com.codality.NotationalFlow
- **Share Ext:** com.codality.NotationalFlow.Share
- **Widgets:** com.codality.NotationalFlow.Widgets

Perhaps for consistency these should look like `com.codality.NotationalFlow.SimplenoteIntents` ?

File: Simplenote.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -4648,6 +4753,181 @@
...
+				MTL_FAST_MATH = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = com.codality.NotationalFlow.Alpha.SimplenoteIntents;
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
+			};
+			name = "Distribution Alpha";
+		};
+		BAB5763126703C8200B0C56F /* Distribution AppStore */ = {
```
Comment (jleandroperez) [inline Simplenote.xcodeproj/project.pbxproj:4901]: Another note (mentioned in another PR). We're probably gonna need to poke the Platform team for help, again, with the provisioning profiles here.

File: Simplenote/Simplenote-Info.plist [hunk 3]
```diff
@@ -74,6 +74,7 @@
 	<string>This app does not require Photo Library access.</string>
 	<key>NSUserActivityTypes</key>
 	<array>
+		<string>SPNoteWidgetIntent</string>
```
Comment (jleandroperez) [inline Simplenote/Simplenote-Info.plist:77]: Let's drop the `SP` prefix from everywhere :fire:

File: SimplenoteIntents/Info.plist [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	<dict>
+		<key>NSExtensionAttributes</key>
+		<dict>
+			<key>IntentsRestrictedWhileLocked</key>
+			<array/>
+			<key>IntentsRestrictedWhileProtectedDataUnavailable</key>
+			<array/>
+			<key>IntentsSupported</key>
+			<array>
+				<string>SPNoteWidgetIntent</string>
```
Comment (jleandroperez) [inline SimplenoteIntents/Info.plist]: Same note as above, let's drop the `SP` prefix

File: SimplenoteIntents/IntentHandler.swift [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+extension IntentHandler: SPNoteWidgetIntentHandling {
+    func provideNoteOptionsCollection(for intent: SPNoteWidgetIntent, with completion: @escaping (INObjectCollection<SPNote>?, Error?) -> Void) {
+        // This is placeholder code to confirm the dynamic intent selection is working
+        // TODO: add logic to fetch the available notes for account
+
+        let placeholderNoteData = ["Note 1", "Note 2", "Note 3", "Note 4"]
+
+        var notes: [SPNote] = []
```
Comment (jleandroperez) [inline SimplenoteIntents/IntentHandler.swift]: Can't find where `SPNote` is defined, but can we rename that to `IntentNote` or `WidgetNote` or so?

**Review by jleandroperez — COMMENTED** [review]
Sending few notes!! First pass looking great!

**Comment by jleandroperez** [comment]
Oh! We also need to handle the "Main App is not logged" scenario. No need to address that in this particular PR, just leaving a note, so that we don't loose track of it.

**Comment by charliescheer** [comment]
Most of the work on this PR was done in #1344  merging this back into the feature branch

File: SimplenoteWidgets/Providers/NotePreviewWidgetProvider.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+struct NotePreviewWidgetProvider: IntentTimelineProvider {
+    typealias Intent = NoteWidgetIntent
+    typealias Entry = NotePreviewWidgetEntry
+
+    func placeholder(in context: Context) -> NotePreviewWidgetEntry {
+        return NotePreviewWidgetEntry(date: Date(), text: "Placeholder")
+    }
+
+    func getSnapshot(for configuration: NoteWidgetIntent, in context: Context, completion: @escaping (NotePreviewWidgetEntry) -> Void) {
+        let entry = NotePreviewWidgetEntry(date: Date(), text: "Placeholder")
```
Comment (jleandroperez) [inline SimplenoteWidgets/Providers/NotePreviewWidgetProvider.swift:17]: Not locallized string

File: SimplenoteWidgets/Providers/NotePreviewWidgetProvider.swift [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    let date: Date
+    let text: String
+}
+
+struct NotePreviewWidgetProvider: IntentTimelineProvider {
+    typealias Intent = NoteWidgetIntent
+    typealias Entry = NotePreviewWidgetEntry
+
+    func placeholder(in context: Context) -> NotePreviewWidgetEntry {
+        return NotePreviewWidgetEntry(date: Date(), text: "Placeholder")
```
Comment (jleandroperez) [inline SimplenoteWidgets/Providers/NotePreviewWidgetProvider.swift:13]: Not localized string

**Review by jleandroperez — COMMENTED** [review]
Hey there @charliescheer !!

Found a couple strings that aren't localized (and I belive those aren't SwiftUI code)

**Pull request merged**